File: backend/src/main/java/com/leavemarker/entity/Subscription.java
package com.leavemarker.entity;

import com.leavemarker.enums.BillingCycle;
import com.leavemarker.enums.SubscriptionStatus;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "subscriptions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Subscription extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "plan_id", nullable = false)
    private Plan plan;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private SubscriptionStatus status = SubscriptionStatus.ACTIVE;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BillingCycle billingCycle;

    @Column(nullable = false)
    private LocalDateTime startDate;

    @Column(nullable = false)
    private LocalDateTime endDate;

    @Column(nullable = false)
    private LocalDateTime currentPeriodStart;

    @Column(nullable = false)
    private LocalDateTime currentPeriodEnd;

    @Column(precision = 10, scale = 2)
    @Builder.Default
    private BigDecimal amount = BigDecimal.ZERO;

    @Column(nullable = false)
    @Builder.Default
    private Boolean autoRenew = true;

    @Column(nullable = false)
    @Builder.Default
    private Boolean isPaid = false;

    // Add-on flags
    @Column(nullable = false)
    @Builder.Default
    private Boolean hasReportDownloadAddon = false;

    @Column(precision = 10, scale = 2)
    @Builder.Default
    private BigDecimal reportDownloadAddonPrice = BigDecimal.ZERO;

    @Column(length = 500)
    private String cancellationReason;

    private LocalDateTime cancelledAt;

    @Column(length = 500)
    private String notes;
}

File: backend-go/internal/models/subscription.go
package models

import (
	"time"

	"github.com/leavemarker/backend-go/internal/enums"
	"github.com/shopspring/decimal"
)

// Subscription represents a subscription entity
type Subscription struct {
	BaseModel
	CompanyID              uint                     `gorm:"not null;index" json:"companyId"`
	PlanID                 uint                     `gorm:"not null;index" json:"planId"`
	Status                 enums.SubscriptionStatus `gorm:"size:20;default:ACTIVE" json:"status"`
	BillingCycle           enums.BillingCycle       `gorm:"size:20" json:"billingCycle"`
	StartDate              time.Time                `json:"startDate"`
	EndDate                time.Time                `json:"endDate"`
	CurrentPeriodStart     time.Time                `json:"currentPeriodStart"`
	CurrentPeriodEnd       time.Time                `json:"currentPeriodEnd"`
	Amount                 decimal.Decimal          `gorm:"type:decimal(10,2);default:0.00" json:"amount"`
	AutoRenew              bool                     `gorm:"default:true" json:"autoRenew"`
	IsPaid                 bool                     `gorm:"default:false" json:"isPaid"`
	HasReportDownloadAddon bool                     `gorm:"default:false" json:"hasReportDownloadAddon"`
	ReportDownloadAddonPrice decimal.Decimal        `gorm:"type:decimal(10,2);default:0.00" json:"reportDownloadAddonPrice"`
	CancellationReason     *string                  `gorm:"size:500" json:"cancellationReason"`
	CancelledAt            *time.Time               `json:"cancelledAt"`
	Notes                  *string                  `gorm:"size:500" json:"notes"`

	// Relationships
	Company  Company   `gorm:"foreignKey:CompanyID" json:"-"`
	Plan     Plan      `gorm:"foreignKey:PlanID" json:"-"`
	Payments []Payment `gorm:"foreignKey:SubscriptionID" json:"-"`
}

func (Subscription) TableName() string {
	return "subscriptions"
}
